extensions-contrib/kafka-emitter/src/main/java/org/apache/druid/emitter/kafka/KafkaEmitter.java

```diff
@@ -170,33 +189,83 @@ public void emit(final Event event)
       Map<String, Object> result = resultBuilder.build();
 
       try {
-        String resultJson = jsonMapper.writeValueAsString(result);
-        ObjectContainer<String> objectContainer = new ObjectContainer<>(
-            resultJson,
-            StringUtils.toUtf8(resultJson).length
+        byte[] resultBytes = jsonMapper.writeValueAsBytes(result);
+        ObjectContainer<byte[]> objectContainer = new ObjectContainer<>(
+            resultBytes,
+            resultBytes.length
         );
         if (event instanceof ServiceMetricEvent) {
-          if (!metricQueue.offer(objectContainer)) {
+          if (!config.getEventTypes().contains(KafkaEmitterConfig.EventType.METRICS) || !metricQueue.offer(objectContainer)) {
```

Comment:
nit, similar comment about avoiding repeated calls to `config.getEventTypes()` and adding the import for `KafkaEmitterConfig.EventType` to shorten the lines and improve readability